File: 数仓表结构/asset_package/src/main/java/com/weshare/util/FileCheckUtils.java
package com.weshare.util;

import com.weshare.core.AssetFileToHive;
import org.apache.commons.lang3.StringUtils;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by mouzwang on 2020-11-30 18:10
 */
public class FileCheckUtils {
    /**
     * check first arg : basic file name
     * @param fileName
     */
    public static void checkFileName(String fileName) {
        if (fileName == null) {
            System.out.println("the file name is null");
            return;
        }
        if (!fileName.equals("project_info") && ! fileName.equals("project_due_bill_no") && !fileName.equals("bag_info")
                 && !fileName.equals("bag_due_bill_no")){
            System.out.println("an illegal file name!");
            return;
        }
    }

    /**
     * check second arg : project id or bag id
     * @param fileId
     */
    public static void checkFileId(String fileId) {
        if (StringUtils.isBlank(fileId)) {
            System.out.println("the project id or bag id is not correct!");
            return;
        }
    }

}

File: 数仓表结构/asset_package/src/main/scala/com/weshare/util/FileCheckUtils.scala
package com.weshare.util

/**
  * @author ximing.wei 2021-05-27 14:26:08
  */
object FileCheckUtils {
    private val fileNameList: List[String] = List("project_info", "project_due_bill_no", "bag_info", "bag_due_bill_no")
    private val rowTypeList: List[String] = List("insert", "delete", "update")

    /**
      * 校验文件名
      *
      * @param fileName 文件名
      * @return 成功返回 true ，失败抛出异常
      */
    def checkFileName(fileName: String): Boolean = {
        if (fileName == null || fileName.trim == "") throw new RuntimeException("文件名参数不应该为空！")
        if (!fileNameList.contains(fileName)) throw new RuntimeException("文件名参数（" + fileName + "）不符合要求" + fileNameList + "！")
        true
    }

    /**
      * 校验项目编号
      *
      * @param fileId 项目编号
      * @return 成功返回 true ，失败抛出异常
      */
    def checkFileId(fileId: String): Boolean = {
        if (fileId == null || fileId.trim == "") throw new RuntimeException("项目编号参数不应该为空！")
        true
    }

    /**
      * 校验操作类型
      *
      * @param rowType 操作类型
      * @return 成功返回 true ，失败抛出异常
      */
    def checkRowType(rowType: String): Boolean = {
        if (rowType == null || rowType.trim == "") throw new RuntimeException("操作类型参数不应该为空！")
        if (!rowTypeList.contains(rowType)) throw new RuntimeException("操作类型参数（" + rowType + "）不符合要求" + rowTypeList + "！")
        true
    }
}
